Source/monster.cpp

```diff
@@ -1116,7 +1116,8 @@ void StartMonsterDeath(int i, int pnum, bool sendmsg)
 
 	Direction md = pnum >= 0 ? GetMonsterDirection(monster) : monster._mdir;
 	NewMonsterAnim(monster, MonsterGraphic::Death, md, gGameLogicStep < GameLogicStep::ProcessMonsters ? AnimationDistributionFlags::ProcessAnimationPending : AnimationDistributionFlags::None);

```

Comment:
looks like `NewMonsterAnim` should also only be called if `_mmode != MonsterMode::Petrified`

Reply:
After I look again at this it loks like this:
Calls to `StartMonsterDeath`:
- `M_StartKill`  - had no petrify check, but there were such checks in function that called it, which I removed.
- `M_SyncStartKill` - had petrify check

seems StartMonsterDeath ALWAYS set `_mode` to `MonsterMode::Death` and calls `	NewMonsterAnim` which sets 
`AnimInfo.IsPetrified` to `true`.

That would mean - that those checks are always redundant. Therefore I remove this condition also for 
```
 if (monster._mmode != MonsterMode::Petrified)
		monster._mmode = MonsterMode::Death;
```
and leave just 
```
monster._mmode = MonsterMode::Death;
```
as it was before.
Is it good for you?

Reply:
Considering how close you are to having this all solved in a really nice way i think that would be a little sad :)

Reply:
> That would mean - that those checks are always redundant. Therefore I remove this condition also for

They where never redundant before this PR as they happened before the call to `StartMonsterDeath`, only after you moved them to after the call did they become ineffective. Which is why you will need to make `NewMonsterAnim()` and `monster._mmode = MonsterMode::Death;` inside `StartMonsterDeath` conditional to not break stone curse.

Reply:
Yes you are right, I will try to fix that :) sorry, I am getting experience right now